app/helpers/application_helper.rb

```diff
@@ -79,8 +79,8 @@ def multi_conditional_field(f, property, enablers)
       label_html: { class: 'conditional-group' }
   end
 
-  def confirm_deleting(record)
-    if ['assignments', 'journals'].include? controller_name
+  def confirm_deleting(record, controller_name)
```

Comment:
But this way you'd need to add a second parameter to every other occasion confirm_deleting is used.
I think something in that manner would be nicer and you don't need to pass controller_name on every other use:
```ruby
def confirm_deleting(record, controller_translate = nil)
  controller_translate ||= controller_name
  if ['assignments', 'journals', 'hours'].include? controller_translate
    confirm_text = t("delete_#{controller_translate.singularize}")
  end
  { method: :delete, data: { confirm: confirm_text || t_confirm_delete(record) } }
end
```